### ISSUE TITLE
Applying display settings fails with kernel 4.17.2

### ISSUE THREAD

**Issue by Saduff** [linuxdeepin/dde-daemon#50]
Hi, after upgrading to kernel 4.17.2, the Deepin daemon fails when applying display settings. It falls back to default settings. Using Deepin daemon v3.2.20.

Found these entries in the boot log:

```
com.deepin.daemon.Display[856]: manager.go:532: [updateMonitors] add monitor: eDP-1 0 0 1920 1080
com.deepin.daemon.Display[856]: manager.go:532: [updateMonitors] add monitor: DP-2-2 1920 0 1920 1080
com.deepin.daemon.Display[856]: manager.go:532: [updateMonitors] add monitor: DP-2-3 3840 0 1920 1080
com.deepin.daemon.Display[856]: manager.go:170: Try apply settings failed for init: exit status 1
                                  ->  display.go:38
                                  ->  main.go:104
                                  ->  asm_amd64.s:2361
```

**Comment by jouyouyun** [comment]
@electricface

**Comment by DNAngeluS** [comment]
Hi!,
same case here I'm using a Razer Blade 14' Late 2016 model, Nvidia GPU and Intel GPU are both working and if you apply display configurations using xrandr from terminal they work. 
ex:
`xrandr --output eDP-1-1 --primary --auto --rotate normal --pos 0x0 --output HDMI-0 --auto --above eDP-1-1 --panning 1920x1080+0+0`
Here are more details about my configuration and errors.

uname -a
`Linux dnangelus-blade 4.17.3-1-ARCH #1 SMP PREEMPT Tue Jun 26 04:42:36 UTC 2018 x86_64 GNU/Linux`

lspci
```
00:00.0 Host bridge: Intel Corporation Xeon E3-1200 v5/E3-1500 v5/6th Gen Core Processor Host Bridge/DRAM Registers (rev 07)
00:01.0 PCI bridge: Intel Corporation Xeon E3-1200 v5/E3-1500 v5/6th Gen Core Processor PCIe Controller (x16) (rev 07)
00:02.0 VGA compatible controller: Intel Corporation HD Graphics 530 (rev 06)
00:08.0 System peripheral: Intel Corporation Xeon E3-1200 v5/v6 / E3-1500 v5 / 6th/7th Gen Core Processor Gaussian Mixture Model
00:14.0 USB controller: Intel Corporation Sunrise Point-H USB 3.0 xHCI Controller (rev 31)
00:14.2 Signal processing controller: Intel Corporation Sunrise Point-H Thermal subsystem (rev 31)
00:15.0 Signal processing controller: Intel Corporation Sunrise Point-H Serial IO I2C Controller #0 (rev 31)
00:15.1 Signal processing controller: Intel Corporation Sunrise Point-H Serial IO I2C Controller #1 (rev 31)
00:16.0 Communication controller: Intel Corporation Sunrise Point-H CSME HECI #1 (rev 31)
00:17.0 SATA controller: Intel Corporation Sunrise Point-H SATA Controller [AHCI mode] (rev 31)
00:1c.0 PCI bridge: Intel Corporation Sunrise Point-H PCI Express Root Port #1 (rev f1)
00:1c.5 PCI bridge: Intel Corporation Sunrise Point-H PCI Express Root Port #6 (rev f1)
00:1d.0 PCI bridge: Intel Corporation Sunrise Point-H PCI Express Root Port #9 (rev f1)
00:1d.4 PCI bridge: Intel Corporation Sunrise Point-H PCI Express Root Port #13 (rev f1)
00:1e.0 Signal processing controller: Intel Corporation Sunrise Point-H Serial IO UART #0 (rev 31)
00:1f.0 ISA bridge: Intel Corporation Sunrise Point-H LPC Controller (rev 31)
00:1f.2 Memory controller: Intel Corporation Sunrise Point-H PMC (rev 31)
00:1f.3 Audio device: Intel Corporation Sunrise Point-H HD Audio (rev 31)
00:1f.4 SMBus: Intel Corporation Sunrise Point-H SMBus (rev 31)
01:00.0 VGA compatible controller: NVIDIA Corporation GP106M [GeForce GTX 1060 Mobile] (rev a1)
3b:00.0 Network controller: Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter (rev 32)
3d:00.0 Non-Volatile memory controller: Samsung Electronics Co Ltd NVMe SSD Controller SM951/PM951 (rev 01)
```

xrandr
```
Screen 0: minimum 8 x 8, current 1920 x 2160, maximum 32767 x 32767
HDMI-0 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 510mm x 290mm
   1920x1080     60.00*+  59.94    50.00    60.05    60.00    50.04  
   1680x1050     59.95  
   1600x900      60.00  
   1440x900      59.89  
   1400x1050     59.98  
   1280x1024     60.02  
   1280x800      59.81  
   1280x720      60.00    59.94    50.00  
   1152x864      60.00  
   1024x768      60.00  
   800x600       60.32  
   720x576       50.00  
   720x480       59.94  
   640x480       59.94    59.93  
eDP-1-1 connected primary 1920x1080+0+1080 (normal left inverted right x axis y axis) 309mm x 173mm
   1920x1080     60.05*+  60.01    59.97    59.96    59.93  
   1680x1050     59.95    59.88  
   1400x1050     59.98  
   1600x900      59.99    59.94    59.95    59.82  
   1280x1024     60.02  
   1400x900      59.96    59.88  
   1280x960      60.00  
   1440x810      60.00    59.97  
   1368x768      59.88    59.85  
   1280x800      59.99    59.97    59.81    59.91  
   1280x720      60.00    59.99    59.86    59.74  
   1024x768      60.04    60.00  
   960x720       60.00  
   928x696       60.05  
   896x672       60.01  
   1024x576      59.95    59.96    59.90    59.82  
   960x600       59.93    60.00  
   960x540       59.96    59.99    59.63    59.82  
   800x600       60.00    60.32    56.25  
   840x525       60.01    59.88  
   864x486       59.92    59.57  
   700x525       59.98  
   800x450       59.95    59.82  
   640x512       60.02  
   700x450       59.96    59.88  
   640x480       60.00    59.94  
   720x405       59.51    58.99  
   684x384       59.88    59.85  
   640x400       59.88    59.98  
   640x360       59.86    59.83    59.84    59.32  
   512x384       60.00  
   512x288       60.00    59.92  
   480x270       59.63    59.82  
   400x300       60.32    56.34  
   432x243       59.92    59.57  
   320x240       60.05  
   360x202       59.51    59.13  
   320x180       59.84    59.32  
DP-1-1 disconnected (normal left inverted right x axis y axis)
HDMI-1-1 disconnected (normal left inverted right x axis y axis)
  1680x1050 (0x4b) 146.250MHz -HSync +VSync
        h: width  1680 start 1784 end 1960 total 2240 skew    0 clock  65.29KHz
        v: height 1050 start 1053 end 1059 total 1089           clock  59.95Hz
  1280x1024 (0x52) 108.000MHz +HSync +VSync
        h: width  1280 start 1328 end 1440 total 1688 skew    0 clock  63.98KHz
        v: height 1024 start 1025 end 1028 total 1066           clock  60.02Hz
  1280x800 (0x5c) 83.500MHz -HSync +VSync
        h: width  1280 start 1352 end 1480 total 1680 skew    0 clock  49.70KHz
        v: height  800 start  803 end  809 total  831           clock  59.81Hz
  1024x768 (0x63) 65.000MHz -HSync -VSync
        h: width  1024 start 1048 end 1184 total 1344 skew    0 clock  48.36KHz
        v: height  768 start  771 end  777 total  806           clock  60.00Hz
  800x600 (0x72) 40.000MHz +HSync +VSync
        h: width   800 start  840 end  968 total 1056 skew    0 clock  37.88KHz
        v: height  600 start  601 end  605 total  628           clock  60.32Hz
  640x480 (0x7f) 25.175MHz -HSync -VSync
        h: width   640 start  656 end  752 total  800 skew    0 clock  31.47KHz
        v: height  480 start  490 end  492 total  525           clock  59.94Hz
```
After trying to change monitor configuration (extended-duplicate-hdmi-internal)
journal -xae
```
<info> utils.go:144: Change output "eDP-1-1" brightness to 1.00 done!
<info> utils.go:139: Change output "HDMI-0" brightness to 1.00
utils.go:144: Change output "eDP-1-1" brightness to 1.00 done!
utils.go:139: Change output "HDMI-0" brightness to 1.00
<info> utils.go:144: Change output "HDMI-0" brightness to 1.00 done!
utils.go:144: Change output "HDMI-0" brightness to 1.00 done!
<error> utils.go:106: Failed to get self active: Caller does not belong to any known session
  ->  manager.go:97
  ->  manager.go:85
  ->  auto.go:58
  ->  signal.go:59
  ->  signal.go:41
  ->  asm_amd64.s:2361
utils.go:106: Failed to get self active: Caller does not belong to any known session
  ->  manager.go:97
  ->  manager.go:85
  ->  auto.go:58
  ->  signal.go:59
  ->  signal.go:41
  ->  asm_amd64.s:2361
show osd "SwitchMonitors"
show osd "SwitchMonitors"
show osd "SwitchMonitors"
show osd "SwitchMonitors"
show osd "SwitchMonitors"
manager.go:328: [switchToOnlyOne] apply (xrandr  --output HDMI-0 --off --output eDP-1-1 --primary --mode 1920x1080 --rate 60 --pos 0x0 --rotate normal --reflect normal) failed: exit status 1
  ->  manager_ifc.go:97
  ->  asm_amd64.s:574
  ->  value.go:447
  ->  value.go:308
  ->  handle_call.go:166
  ->  handle_call.go:222
  ->  asm_amd64.s:2361
manager_ifc.go:112: Switch mode to 3 failed: exit status 1
  ->  asm_amd64.s:574
  ->  value.go:447
  ->  value.go:308
  ->  handle_call.go:166
  ->  handle_call.go:222
  ->  asm_amd64.s:2361
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 6240, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 6245, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 2657, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 2662, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 5086, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 5091, resource id: 0, major code: 140 (Unknown), minor code: 20
qt.qpa.xcb: QXcbConnection: XCB error: 148 (Unknown), sequence: 1031, resource id: 0, major code: 140 (Unknown), minor code: 20
qt.qpa.xcb: QXcbConnection: XCB error: 148 (Unknown), sequence: 1036, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 1469, resource id: 0, major code: 140 (Unknown), minor code: 20
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 148 (Unknown), sequence: 1474, resource id: 0, major code: 140 (Unknown), minor code: 20
```

**Comment by jouyouyun** [comment]
From the log `xrandr  --output HDMI-0 --off --output eDP-1-1 --primary --mode 1920x1080 --rate 60 --pos 0x0 --rotate normal --reflect normal` found the rate is wrong.
@electricface try to fix it.

**Comment by electricface** [comment]
https://cr.deepin.io/#/c/dde/startdde/+/36601

**Comment by Saduff** [comment]
Fixed in v3.2.21.
